ui/components/src/model/units/percent.ts

```diff
@@ -44,12 +43,28 @@ export const PERCENT_UNIT_CONFIG: Readonly<Record<PercentUnitKind, UnitConfig>>
   },
 };
 
-export function formatPercent(value: number, unitOptions: PercentUnitOptions): string {
-  const decimals = unitOptions.decimal_places ?? DEFAULT_DECIMAL_PLACES;
+export function formatPercent(value: number, { kind, decimal_places }: PercentUnitOptions): string {
+  // Intl.NumberFormat translates 0 -> 0%, 0.5 -> 50%, 1 -> 100%
+  if (kind === 'Percent') {
+    value = value / 100;
+  }
+
+  const hasDecimalPlaces = decimal_places !== undefined;
 
-  if (unitOptions.kind === 'PercentDecimal') {
-    value = value * 100;
+  let formatter;
+  if (hasDecimalPlaces) {
+    formatter = new Intl.NumberFormat('en-US', {
+      style: 'percent',
+      maximumFractionDigits: decimal_places,
+      useGrouping: true,
+    });
+  } else {
+    formatter = new Intl.NumberFormat('en-US', {
+      style: 'percent',
+      maximumSignificantDigits: 3,
+      useGrouping: true,
+    });
```

Comment:
It looks like the only diff between these is `maximumSignificantDigits`. I could see this being easier to read/maintain if you do something like `const decimalPlaces = decimal_places !== undefined ? decimal_places : 3` and then use that var as the value for `maximumSignificantDigits` in a single formatter.

Reply:
Ah it's actually that one is using maximum**Fraction**Digits and the other is using maximum**Significant**Digits

Do you see any way to make that more clear? (it is definitely not obvious at a glance)

Reply:
I guess when all else fails, I could leave a comment! Do you think that would be nice?

Reply:
I don't mind the `if / else` at a quick glance, but yeah a comment could be nice with a quick explanation of maximumFractionDigits and maximumSignificantDigits!

Reply:
Ahhh, I totally missed those properties were slightly different on a skim. I think if you leave it as is, it might be nice to pull that `3` out into a variable named `DEFAULT_DECIMAL_PLACES` or something like that. I think that helps a lot with making those types of values easier to find and modify when needed.

Reply:
That said, I also think it's fine as is. It's still very readable.

Reply:
Updated, I like the `const` idea too